### PR TITLE
fix: no focus border on date cells

### LINKED ISSUE #1414: [Bug] No focus border on date cells
### Bug Description

Title says all.

### How to Reproduce

1. Go to a grid with a date property
2. Click on a cell under the date property
3. Observe that there is no focus border

### Expected Behavior

Cell should have a bright blue focus border

### Operating System

Fedora Linux 36

### AppFlowy Version(s)

v0.0.6

### Screenshots

image

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Eakam1007**
Fixes #1414

Added a focus border to date cells if the popup is open:

image

### CODE REVIEW

File: frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart [hunk 1]
```diff
@@ -71,19 +72,30 @@ class _DateCellState extends GridCellState<GridDateCell> {
             margin: EdgeInsets.zero,
             child: SizedBox.expand(
               child: GestureDetector(
-                behavior: HitTestBehavior.opaque,
-                onTap: () => _popover.show(),
```
Comment (richardshiue) [inline frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart:75]: Hello @Eakam1007, thanks for contributing! Your fix works well, but the `GridCellWidget` class already  provides this functionality. It just hasn't been hooked up yet. It should be enough to simply add a line to set `widget.onCellEditing.value` to `true` in this `onTap` handler.

Comment (Eakam1007) [reply frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart:75]: Oh, I missed that. I will update it, thanks

Comment (Eakam1007) [reply frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart:75]: I tried updating it and it works for a new cell. However, if you change the type of the cell to something else (such as text) and then back, the border stops showing.
This also happens for cells of type `Select` and `URL` where it is implemented in a similar way. Is this a known bug? I could not find an issue for this

Comment (richardshiue) [reply frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart:75]: Yeah I can also reproduce this. Could you open a separate issue for that?

Comment (Eakam1007) [reply frontend/app_flowy/lib/plugins/grid/presentation/widgets/cell/date_cell/date_cell.dart:75]: Sure, I will update this branch and open a separate issue for this

**Review by richardshiue — APPROVED** [review]
LGTM!

**Review by richardshiue — CHANGES_REQUESTED** [review]
Hello @Eakam1007, I missed the fact that the GridCellState itself already has a method called `requestBeginFocus()` that is hooked up to listen for focus changes. You can move those two lines from `onTap()` to the override implementation below. After doing this, you can also get rid of the `GestureDetector` so that the immediate child of `SizedBox.expand` is the `Align`. Sorry for the mixup

**Review by richardshiue — APPROVED** [review]
LGTM, for real this time. Thanks for contributing @Eakam1007!

**Pull request merged**